vtm/src/org/oscim/renderer/ExtrusionRenderer.java

```diff
@@ -45,11 +46,15 @@
     protected float mAlpha = 1;
 
     private float mZLimit = Float.MAX_VALUE;
-    private float[] mLightPos = new float[]{0.3f, 0.3f, 1f};
+
+    private Sun mSun;
+    private boolean mEnableCurrentSunPos = false;
 
     public ExtrusionRenderer(boolean mesh, boolean translucent) {
         mMesh = mesh;
         mTranslucent = translucent;
+
+        mSun = new Sun();
```

Comment:
Quick comment: since the feature is optional, better have a lazy initialization of any feature objects, specially if they carry collections.

Reply:
You mean lazy init the sun, or the collection?
Probably can achieve lazy init collection when calling `updateColor()` the first time.
Yes it's a feature, but fully customizable. Storing in another attribute next to the sun would be confusing.

Reply:
The `Sun` class.
Can use `getSun` everywhere and if null at first call, just create it there.

Reply:
But we have to specify light position in any case?

Reply:
Isn't the feature optional and use a default position as is now in master?

Reply:
All parameters in sun are optional. So why make an option of an option, which in fact gives the same results and only makes it more complicated.

Yes this whole construct is needed for shadows, too. But also gives new options for extrusion renderer only.

Reply:
I don't think storing 5 values more has impact on performance or memory. It's a more physical based way to specify lights position.

Reply:
Let's wait to be reviewed then. Most concern is with users who simply don't care for light or shadows and want to continue with same fast rendering without new possible GL issues.

Reply:
I don't see any regression here. If somebody don't care for new features, can stay on tested versions.